### PR TITLE
Improve `toc-depth` in html output

### PR DESCRIPTION
**Pull request by dragonstyle**
The `toc-depth` option will now control how deep html table of contents will display (previously, we always displayed to TOC to a depth of 3). This will control what depth of items are show when the `toc-location` is body. It will control what depth of items will be opened / tracked when scrolling when `toc-location` is left or right.

### CODE REVIEW

File: src/format/html/format-html-bootstrap.ts [hunk 1]
```diff
@@ -275,11 +279,17 @@ function bootstrapHtmlPostprocessor(
       }
 
       // default collapse non-top level TOC nodes
-      const nestedUls = toc.querySelectorAll("ul ul");
-      for (let i = 0; i < nestedUls.length; i++) {
-        const ul = nestedUls[i] as Element;
-        ul.classList.add("collapse");
+      const tocDepth = format.pandoc[kTocDepth] || 3;
```
Comment (dragonstyle) [inline src/format/html/format-html-bootstrap.ts:282]: Is there a better place for me to be defaulting this?

File: changelog.md [hunk 2]
```diff
@@ -1,3 +1,5 @@
 # Changelog
 
 ## Unreleased
+
+- respect `toc-depth` in the HTML format (bootstrap) rather than always acting as if depth is 3.
```
Comment (jjallaire) [inline changelog.md:5]: Changelog should be a separate commit (in case we want to cherry pick back onto v1.0)

File: src/format/html/format-html-bootstrap.ts (hunk 1 above)
Comment (jjallaire) [reply src/format/html/format-html-bootstrap.ts:282]: That seems fine to me

File: changelog.md [hunk 3]
```diff
@@ -1,3 +1,5 @@
 # Changelog
 
 ## Unreleased
```
Comment (jjallaire) [inline changelog.md:3]: Let's remove the changelog entry here
